src/Wizard/Services/ConnectivityService.ts

```diff
@@ -0,0 +1,44 @@
+import { Connectivity } from '../Connectivity/Connectivity';
+import { TestType } from '../Models/ConnectivityTest';
+import { inject, injectable, injectAll } from 'tsyringe';
+import { logger } from 'src/Utils';
+import { URL } from 'url';
+
+export const ConnectivityUrls = Symbol('ConnectivityUrls');
+
+@injectable()
+export class ConnectivityAnalyzer {
+  constructor(
+    @inject(ConnectivityUrls) private readonly options: Map<TestType, URL>,
+    @injectAll(Connectivity)
+    private readonly connectivityTestRegistry: Connectivity[]
+  ) {}
+
+  public async verifyAccess(type: TestType): Promise<boolean> {
+    logger.debug('Calling connectivity status test with type %s', type);
+
+    const connectivity = this.findConnectivity(type);
+
+    return connectivity.test(this.options.get(connectivity.type));
+  }
+
+  public async verifyConnection(url: URL): Promise<boolean> {
+    const connectivity = this.findConnectivity(TestType.HTTP);
+
+    return connectivity.test(url);
+  }
```

Comment:
You can introduce a single method to handle both cases

```suggestion
  public async verifyAccess(type: TestType, url?: URL): Promise<boolean> {
    logger.debug('Calling connectivity status test with type %s', type);

    const connectivity = this.findConnectivity(type);

    return connectivity.test(url ?? this.options.get(connectivity.type));
  }
```